### DISCUSSION TITLE
Does C# have anything equivalent of Java modules?

### DISCUSSION THREAD

**Discussion by siddhsql** [dotnet/runtime#94546] [General]
I am wondering if C# has anything that is equivalent of Java modules [[1](https://www.oracle.com/corporate/features/understanding-java-9-modules.html)]

**Comment by HaloFour** [comment]
.NET has assemblies and `internal` accessibility.

**Comment by siddhsql** [comment]
C# is much better than Java. with a single keyword it addresses most of
what Java accomplishes through a mind-boggling complicated design and
architecture that takes them 5 years or so to deliver.
